packages/icon-button/src/icon-button-base.ts

```diff
@@ -58,18 +66,27 @@ export class IconButtonBase extends BaseElement {
   }
 
   protected handleClick() {
-    if (this.offIcon !== '') {
+    if (this.shouldToggle) {
       this.on = !this.on;
       this.mdcFoundation.handleClick();
     }
   }
 
+  protected calculateShouldToggle() {
+    this.shouldToggle =
+        !(this.offIcon === '' && !this.offIconSlot.assignedNodes().length);
```

Comment:
Mild nitty preference for `this.offIcon !== '' || this.offIconSlot.assignedNodes().length > 0` -- but your call.

Reply:
Done